sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -252,12 +260,12 @@ def sleep(self):
         sleep(self._statement_sleep_seconds)
 
     def refresh_status(self):
-        status = self._http_client.get_session(self.id)[u'state']
+        response = self._http_client.get_session(self.id)
+        status = response[u'state']
+        log_array = response[u'log']
 
-        if status in constants.POSSIBLE_SESSION_STATUS:
-            self.status = status
-        else:
-            raise LivyUnexpectedStatusException(u"Status '{}' not supported by session.".format(status))
```

Comment:
In theory, I can separate it out. But while I am doing this work, I already noticed that there are 2 new Livy state that is not in the list. If I don't add the new states, the test could randomly fail. Those 2 states should not affect how sparkmagic works at all. 
I agree sparkmagic is depending on Livy and should have enough understanding of it. We need to know the ending states: Dead/Success and some key states like Idle. However, it shouldn't have a hard coded list of all Livy states and strictly enforcing them, especially these states are not used by magic anywhere else. Livy could add a new transition state any time, and I don't think we should put magic into a corner where a new update is needed for every such Livy change. It is almost impossible to keep up with that on time. Also, what if Livy removes a state? current way will not guarantee correctness either, so i really don't see a significant value of this assert.